src/components/Language.tsx

```diff
@@ -1,60 +1,206 @@
-import { Button, Container, Stack, TextInput, Text } from '@mantine/core';
+import { Button, Container, TextInput, Text, Modal, MultiSelect, Space, Group, List, Card, Avatar, Chip, Header, Title } from '@mantine/core';
+import { showNotification } from '@mantine/notifications';
 import { LanguageHandle } from '@perspect3vism/ad4m';
-import { useContext, useState } from 'react';
-import { Ad4mContext } from '..';
+import { useContext, useEffect, useState } from 'react';
+import { AgentContext } from '../context/AgentContext';
+import { generateLanguageInitials, isSystemLanguage } from '../util';
+import { MainContainer, MainHeader } from './styles';
 
 type Props = {
 }
 
 const Language = (props: Props) => {
-  const ad4mClient = useContext(Ad4mContext);
-
-  const [languageAddr, setLanguageAddr] = useState("");
-  const [language, setLanguage] = useState<LanguageHandle | null>(null);
-  const [loading, setLoading] = useState(false);
-
-  const getLanguage = async (event: React.SyntheticEvent) => {
-    event.preventDefault();
-    setLoading(true);
-    try {
-      let language = await ad4mClient.languages.byAddress(languageAddr);
-      console.log("language get result, ", language);
-      setLanguage(language);
-    } catch (err) {
-      console.log(err);
+  const {state: {
+    client
+  }} = useContext(AgentContext);
+
+  const [languages, setLanguages] = useState<any[] | null[]>([]);
+  const [installLanguageModalOpen, setInstallLanguageModalOpen] = useState(false);
+
+  const [languageName, setLanguageName] = useState("");
+  const [languageDescription, setLanguageDescription] = useState("");
+  const [languageSourceLink, setLanguageSourceLink] = useState("");
+  const [languageBundlePath, setLanguageBundlePath] = useState("");
+  const [data, setData] = useState<any[]>([]);
+
+  const installLanguage = async () => {
+    if (languageBundlePath) {
+      const installedLanguage = await client!.languages.publish(languageBundlePath, {
```

Comment:
I probably miss writing the issue in language page, when I want a `install language feature`, I'm more meaning install a language given its hash, which ad4min will get this language and install it, and `get language feature` is just get the language meta information. Here you have `publish language feature`, I think it's also good to have, just don't forget other two features.